src/lib.rs

```diff
@@ -0,0 +1,46 @@
+#[macro_use]
+extern crate lazy_static;
+extern crate rand;
+
+use rand::Rng;
+use rand::distributions::{Normal, IndependentSample};
+
+///
+/// A rand::Rng implementation which is safe for wasm-unknown-unknown.
+///
+/// Requires `javascript_math_random` to be an import identifier corresponding
+/// to the browser's JavaScript Math.random function.
+///
+pub struct JavaScriptRng;
+
+extern "C" {
+    fn javascript_math_random() -> f64;
+}
+
+impl Rng for JavaScriptRng {
+    fn next_u32(&mut self) -> u32 {
+        unsafe {
+            // The maximum value of a u32 is 4,294,967,295. We go one over here
+            // because the JavaScript Math.random call returns values between
+            // 0 (inclusive) and 1 (exclusive). Without the + 1 here, the floor
+            // call would mean the maximum u32 value couldn't be returned.
+            let upper: i64 = 4294967295 + 1;
```

Comment:
Maybe use the underscore syntax here for readability? `4_294_967_295`? Why not `<u32>::max_value()` or `std::u32::MAX`?

Reply:
Not `std::u32::MAX` since needs a cast to `i64` then. The underscore syntax is nice.